### PR TITLE
enhance the volcano topology framework

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>
issue #1763
the PR is consists of the following two parts:
1. expand the taskInfo to  add a member NumaInfo *TopologyInfo to save the pod topology allocated data
2. add a annotation key : volcano.sh/topology-decision to submit the the pod topology allocated data to kubelet

### CODE REVIEW

**Comment by k82cn** [comment]
Any background about this PR?

**Comment by justadogistaken** [comment]
What we are going to do with "PodResourceDecision"? Trying to apply allocate decistion to container in the future? Or update current resource context for a scheduling perioid.

File: pkg/scheduler/api/numa_info.go [hunk 1]
```diff
@@ -118,6 +141,50 @@ func (info *NumatopoInfo) Release(resSets ResNumaSets) {
...
+	decision := PodResourceDecision{}
+	err := json.Unmarshal([]byte(ti.Pod.Annotations[topologyDecisionAnnotation]), &decision)
+	if err != nil {
+		return nil
+	}
+
+	return decision.NUMAResources
+}
+
+func (info *NumatopoInfo) AddTask(ti *TaskInfo) {
```
Comment (william-wang) [inline pkg/scheduler/api/numa_info.go:162]: please add the function description for all new added functions like `AddTask`, `RemoveTask`.

File: pkg/scheduler/api/pod_info.go [hunk 2]
```diff
@@ -117,14 +118,27 @@ func GetPodRevocableZone(pod *v1.Pod) string {
 	return ""
 }
 
-// GetPodTopologyPolicy return volcano.sh/numa-topology-policy value for pod
-func GetPodTopologyPolicy(pod *v1.Pod) string {
+// GetPodTopologyInfo return volcano.sh/numa-topology-policy value for pod
```
Comment (william-wang) [inline pkg/scheduler/api/pod_info.go:121]: What's the meaning of volcano.sh/numa-topology-policy value? Does it contains the policy on pod and the allocated numa resource?

File: pkg/scheduler/api/resource_info.go [hunk 3]
```diff
@@ -85,6 +85,32 @@ func NewResource(rl v1.ResourceList) *Resource {
...
 }
 
+// ResFloat642Quantity transform resource quantity
+func ResFloat642Quantity(resName v1.ResourceName, quantity float64) resource.Quantity {
+	var resQuantity *resource.Quantity
+	switch resName {
+	case v1.ResourceCPU:
+		resQuantity = resource.NewMilliQuantity(int64(quantity), resource.DecimalSI)
+	default:
+		resQuantity = resource.NewQuantity(int64(quantity), resource.BinarySI)
```
Comment (william-wang) [inline pkg/scheduler/api/resource_info.go:95]: does the default works for memory? or will have to add a memory case for it.

File: pkg/scheduler/api/job_info.go [hunk 4]
```diff
@@ -80,6 +81,24 @@ func (ctx *TransactionContext) Clone() *TransactionContext {
 	return &clone
 }
 
+type TopologyInfo struct {
+	Policy string
+	ResMap map[int]v1.ResourceList
```
Comment (william-wang) [inline pkg/scheduler/api/job_info.go:86]: add the description for key.

File: pkg/scheduler/api/numa_info.go [hunk 5]
```diff
@@ -69,10 +80,22 @@ func (info *NumatopoInfo) DeepCopy() *NumatopoInfo {
...
 		tmpInfo.Allocatable = resInfo.Allocatable.Clone()
-		numaInfo.NumaResMap[resName] = &tmpInfo
+
+		for numaId, data := range resInfo.AllocatablePerNuma {
+			tmpInfo.AllocatablePerNuma[numaId] = data
+		}
+
+		for numaID, data := range resInfo.UsedPerNuma {
+			tmpInfo.UsedPerNuma[numaID] = data
+		}
```
[the remark this answers is not in the document]
Comment (huone1) [reply pkg/scheduler/api/numa_info.go:96]: I think ID is better

File: pkg/scheduler/api/numa_info.go (hunk 1 above)
Comment (huone1) [reply pkg/scheduler/api/numa_info.go:163]: done, added the  function description

File: pkg/scheduler/api/pod_info.go (hunk 2 above)
Comment (huone1) [reply pkg/scheduler/api/pod_info.go:121]: yep, you're right

File: pkg/scheduler/api/resource_info.go (hunk 3 above)
Comment (huone1) [reply pkg/scheduler/api/resource_info.go:95]: yes ,  the default works for memory by resource.NewQuantity

File: pkg/scheduler/api/job_info.go (hunk 4 above)
Comment (huone1) [reply pkg/scheduler/api/job_info.go]: done, added it

**Comment by huone1** [comment]
> What we are going to do with "PodResourceDecision"? Trying to apply allocate decistion to container in the future? Or update current resource context for a scheduling perioid.

yes, it is better if the kubelet supports to use it to allocate the resource

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**